src/lava/lib/peripherals/liveplot/liveplot.py

```diff
@@ -0,0 +1,79 @@
+# Copyright (C) 2022 Intel Corporation
+# SPDX-License-Identifier: BSD-3-Clause
+# See: https://spdx.org/licenses/
+
+import cv2
+import numpy as np
+
+from lava.magma.core.process.process import AbstractProcess
+from lava.magma.core.process.ports.ports import InPort
+from lava.magma.core.sync.protocols.loihi_protocol import LoihiProtocol
+from lava.magma.core.model.py.ports import PyInPort
+from lava.magma.core.model.py.type import LavaPyType
+from lava.magma.core.resources import CPU
+from lava.magma.core.decorator import implements, requires
+from lava.magma.core.model.py.model import PyLoihiProcessModel
+
+
+class OpenCVDisplay(AbstractProcess):
+    """ Process that display incoming image with OpenCV.
+
+    This process continuously display grayscale/RGB images coming through its
+    InPort. The number of input ports is decided by the number of process models
+    connected to this opencv process model.
+
+    Parameters
+    ----------
+    shape: tuple
+        Shape of the InPort.
+        (height, width) or (height, width, 1) for grayscale images.
+        (height, width, 3) for RGB images.
+    opencv_window_name: str
+        Name of OpenCV window.
+    """
+    def __init__(self,
+                 shape: tuple,
+                 opencv_window_name: str = "Image Display"):
+        super().__init__(shape=shape,
+                         opencv_window_name=opencv_window_name)
+        if len(shape) not in (2, 3):
+            raise ValueError("Shape must be in the format (height, width) for \
+                             grayscale or (height, width, 3) for RGB.")
+        if len(shape) == 3 and shape[2] not in (1, 3):
+            raise ValueError("For a 3-channel image, shape must be \
+                             (height, width, 3). For grayscale, use \
+                             (height, width) or (height, width, 1).")
+        self.frame_port = InPort(shape=shape)
+
+
+@implements(proc=OpenCVDisplay, protocol=LoihiProtocol)
+@requires(CPU)
+class OpenCVDisplayPM(PyLoihiProcessModel):
```

Comment:
Add OpenCV window destroy function when the PM finishing with the API cv2.destroyWindow()